src/libstd/sys/cloudabi/mutex.rs

```diff
@@ -52,8 +52,8 @@ pub struct ReentrantMutex {
 }
 
 impl ReentrantMutex {
-    pub unsafe fn uninitialized() -> ReentrantMutex {
-        mem::uninitialized()
+    pub unsafe fn uninitialized() -> mem::MaybeUninit<ReentrantMutex> {
```

Comment:
This doesn't seem right, you are changing the API of a public type.

The Windows mutex looked similar; these days it is declared as
```rust
pub struct ReentrantMutex { inner: UnsafeCell<MaybeUninit<c::CRITICAL_SECTION>> }
```
So you'll probably want to adjust the `struct` above here as well.

Reply:
Fixed the API in the latest push, but after reading through the implementation of the ReentrantMutex for cloudabi specifically, I've tried to leave the implementation roughly the same because it seems to use primitives instead of libc features due to cloudabi lacking some libc features.

Reply:
Thanks! Looking much better; I left some notes.